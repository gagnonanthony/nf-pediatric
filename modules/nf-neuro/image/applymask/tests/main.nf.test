nextflow_process {

    name "Test Process IMAGE_APPLYMASK"
    script "../main.nf"
    process "IMAGE_APPLYMASK"

    tag "modules"
    tag "modules_nfcore"
    tag "image"
    tag "image/applymask"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "heavy.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("image - applymask - 4D") {

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/dwi/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi/mask.nii.gz")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("image - applymask - 3D") {

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/anat/anat_image.nii.gz"),
                        file("\${test_data_directory}/anat/anat_mask.nii.gz")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
