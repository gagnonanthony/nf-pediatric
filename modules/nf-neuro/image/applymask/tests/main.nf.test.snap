{
    "image - applymask - 3D": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_masked.nii.gz:md5,2fef128e99cb8bf76e68d5ad7c8f9f6e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,72f62ad02bf476e0defd8c105c4eb83d"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_masked.nii.gz:md5,2fef128e99cb8bf76e68d5ad7c8f9f6e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,72f62ad02bf476e0defd8c105c4eb83d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-25T15:12:20.565448"
    },
    "image - applymask - 4D": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_masked.nii.gz:md5,8c8b58052f2b573d6fa8a6a221405131"
                    ]
                ],
                "1": [
                    "versions.yml:md5,72f62ad02bf476e0defd8c105c4eb83d"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_masked.nii.gz:md5,8c8b58052f2b573d6fa8a6a221405131"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,72f62ad02bf476e0defd8c105c4eb83d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-25T15:11:56.732018"
    }
}